src/utils/dateUtils/dateCalculations.js

```diff
@@ -0,0 +1,18 @@
+export const calculateDateDaysInPast = (date, days) => {
+  const currentDate = date.getDate();
+  const newDate = currentDate - days;
+  date.setDate(newDate);
+  return date;
+};
+
+export const calculateDateDaysInPastFromToday = days => {
+  const todaysDate = new Date();
+  return calculateDateDaysInPast(todaysDate, days);
+};
+
+export const calculateDateMonthsInPast = (date, months) => {
+  const currentMonth = date.getMonth();
+  const newMonth = currentMonth - months;
+  date.setMonth(newMonth);
```

Comment:
Same comment as above:

This modifies the original date (in the calling function as well), and could have unintended side effects. Instead, we should create a copy of the original date and modify+return that one.